src/data/java/com/github/minecraftschurli/arsmagicalegacy/data/AMEnglishLanguageProvider.java

```diff
@@ -0,0 +1,196 @@
+        itemIdTranslation(AMItems.RUNE_BAG);
+        itemIdTranslation(AMItems.ARCANE_COMPOUND);
+        itemIdTranslation(AMItems.ARCANE_ASH);
+        itemIdTranslation(AMItems.PURIFIED_VINTEUM_DUST);
+        blockIdTranslation(AMBlocks.AUM);
+        blockIdTranslation(AMBlocks.CERUBLOSSOM);
+        blockIdTranslation(AMBlocks.DESERT_NOVA);
+        blockIdTranslation(AMBlocks.TARMA_ROOT);
+        blockIdTranslation(AMBlocks.WAKEBLOOM);
+        blockIdTranslation(AMBlocks.VINTEUM_TORCH);
+        blockTranslation(AMBlocks.WIZARDS_CHALK, "Wizard's Chalk");
+        advancementTranslation("root", ArsMagicaLegacy.getModName(), "A renewed look into Minecraft with a splash of magic...");
+    }
+
+    /**
+     * Adds a block translation that matches the block id.
+     *
+     * @param block The block to generate the translation for.
+     */
+    private void blockIdTranslation(Supplier<? extends Block> block) {
+        addBlock(block, idToTranslation(block.get().getRegistryName().getPath()));
+    }
+
+    /**
+     * Adds a block translation.
+     *
+     * @param block       The block to generate the translation for.
+     * @param translation The translation to apply.
+     */
+    private void blockTranslation(Supplier<? extends Block> block, String translation) {
+        addBlock(block, translation);
+    }
+
+    /**
+     * Adds an item translation that matches the item id.
+     *
+     * @param item The item to generate the translation for.
+     */
+    private void itemIdTranslation(Supplier<? extends Item> item) {
+        addItem(item, idToTranslation(item.get().getRegistryName().getPath()));
+    }
+
+    /**
+     * Adds an item translation.
+     *
+     * @param item        The item to generate the translation for.
+     * @param translation The translation to apply.
+     */
+    private void itemTranslation(Supplier<? extends Item> item, String translation) {
+        addItem(item, translation);
+    }
+
+    /**
+     * Adds an effect translation that matches the effect id.
+     *
+     * @param effect The effect to generate the translation for.
+     */
+    private void effectIdTranslation(Supplier<? extends MobEffect> effect) {
+        addEffect(effect, effect.get().getRegistryName().getPath());
+    }
+
+    /**
+     * Adds an enchantment translation that matches the enchantment id.
+     *
+     * @param enchantment The enchantment to generate the translation for.
+     */
+    private void enchantmentIdTranslation(Supplier<? extends Enchantment> enchantment) {
+        addEnchantment(enchantment, enchantment.get().getRegistryName().getPath());
+    }
+
+    /**
+     * Adds an entity translation that matches the entity id.
+     *
+     * @param entity The entity to generate the translation for.
+     */
+    private void entityIdTranslation(Supplier<? extends EntityType<?>> entity) {
+        addEntityType(entity, entity.get().getRegistryName().getPath());
+    }
```

Comment:
- forgot call to idToTranslation
- use RegistryObject directly
```suggestion
    private void entityIdTranslation(RegistryObject<? extends EntityType<?>> entity) {
        addEntityType(entity, idToTranslation(entity.getId().getPath()));
    }
```